src/main/java/Parser.java

```diff
@@ -0,0 +1,152 @@
+import CustomExceptions.*;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class Parser {
+    private final String input;
+    private static final ArrayList<String> validActions
+            = new ArrayList<>(Arrays.asList("todo", "deadline", "event", "done", "delete", "list", "bye"));
```

Comment:
Since this is a constant, it should follow the coding standard of all uppercase and using underscore to separate words.